packages/nameguard-sdk/src/index.ts

```diff
@@ -528,18 +536,29 @@ export class NameGuard {
    * using the network specified in the `NameGuard` instance. Therefore the returned `name` may not match the provided `name`, but is guaranteed to have a matching `namehash`.
    *
    * @param {string} name The name for NameGuard to inspect.
-   * @param {InspectNameOptions} options The options for the inspection.
    * @returns {Promise<NameGuardReport>} A promise that resolves with the `NameGuardReport` of the name. Check the `inspected` field of the result to determine if the result is an `InspectedNameGuardReport` or an `UninspectedNameGuardReport` for performance reasons in the case that the provided `name` was exceptionally long.
    * @example
    * const nameGuardReport = await nameguard.inspectName('vitalik.eth');
    */
-  public inspectName(
-    name: string,
-    options?: InspectNameOptions,
-  ): Promise<NameGuardReport> {
+  public async inspectName(name: string): Promise<NameGuardReport> {
     const network_name = this.network;
 
-    return this.rawRequest("inspect-name", "POST", { name, network_name });
+    try {
+      return await this.rawRequest("inspect-name", "POST", {
+        name,
+        network_name,
+      });
+    } catch (error) {
+      if (error instanceof NameGuardError) {
+        throw error;
+      }
+
+      throw new NameGuardError(
+        NameGuardErrorType.Input,
```

Comment:
This isn't necessarily an input error. Could we add logic here to classify the error type as accurately as possible?

Reply:
@lightwalker-eth I had this locally I didn't push, because it probably needs more thought about the errors returning from the API too so we can throw the correct errors in code.

Here's what I was thinking:

```ts
if (error instanceof TypeError && error.message.includes("Failed to fetch")) {
      throw new NameGuardError(
        NameGuardErrorType.Network,
        undefined,
        "Network error occurred while inspecting the name"
      );
    }

    if (error instanceof Error && error.message.includes("timeout")) {
      throw new NameGuardError(
        NameGuardErrorType.Timeout,
        undefined,
        "Request timed out while inspecting the name"
      );
    }

    if (error instanceof Error && error.message.toLowerCase().includes("invalid")) {
      throw new NameGuardError(
        NameGuardErrorType.Input,
        undefined,
        `Invalid input: ${error.message}`
      );
    }

    // If we can't classify it more specifically, use Unknown
    throw new NameGuardError(
      NameGuardErrorType.Unknown,
      undefined,
      `An unexpected error occurred: ${error instanceof Error ? error.message : String(error)}`
    );
  }
```

Reply:
@notrab Please see our other active comment thread on this PR. My understanding is the other approach you've suggested for inside `rawRequest` would be better than the approach suggested in the message above, and therefore that the approach suggested above could be removed.

If I'm missing something here please let me know :+1: